src/core/control/Control.cpp

```diff
@@ -2394,6 +2397,39 @@ auto Control::openFile(fs::path filepath, int scrollToPage, bool forceOpen) -> b
     return true;
 }
 
+auto Control::changePdfBackground(fs::path filepath, int scrollToPage, bool forceOpen) -> bool {
+
+    LoadHandler loadHandler;
+
+    const fs::path current_filepath = ((*this).doc)->getFilepath();
+    if (current_filepath.empty()) {
```

Comment:
There are some issues with this approach. Whether the PDF background can be changed only depends on whether the `.xopp`-file has been saved at some point or not. 
- There are `.xopp`-files without PDF background and it doesn't make sense to change the PDF background in that case.
- When the document has unsaved changes, these are discarded (without any warning)
- When you have a PDF background (annotate PDF), but it's not saved yet, you can't change the PDF background, although it would make sense.

Reply:
Thank you for the comments :).

* Yes, in fact if you try to do that you get the error "Error in changing the background: please open some file first."
* Fixed with last commit
* Fixed with last commit